mlir/lib/Analysis/PresburgerSet.cpp

```diff
@@ -318,6 +318,52 @@ bool PresburgerSet::findIntegerSample(SmallVectorImpl<int64_t> &sample) {
   return false;
 }
 
+PresburgerSet PresburgerSet::coalesce() {
+  assert(nSym == 0 && "symbolic sets are not yet supported by coalesce");
+  PresburgerSet newSet =
+      PresburgerSet::getEmptySet(this->getNumDims(), this->getNumSyms());
+  ArrayRef<FlatAffineConstraints> basicSetVector =
+      flatAffineConstraints;
+  SmallVector<bool, 4> marked(this->getNumFACs());
+
+  for (unsigned i = 0; i < basicSetVector.size(); i++) {
+    if (marked[i])
+      continue;
+    FlatAffineConstraints bs1 = basicSetVector[i];
+
+    unsigned numIneq = bs1.getNumInequalities();
+    unsigned numEq = bs1.getNumEqualities();
+    // check if bs1 is contained in any basicSet
+    for (unsigned j = 0; j < basicSetVector.size(); j++) {
```

Comment:
Please use `for (unsigned j = 0, e = basicSetVector.size(); j < e; ++j) {` form.